zio-http/src/main/scala/zhttp/service/EncodeRequest.scala

```diff
@@ -13,7 +14,7 @@ trait EncodeRequest {
     val method      = req.method.asJHttpMethod
     val uri         = req.url.path match {
       case Root => "/"
-      case path => path.asString
+      case _    => URL.asUriString(req.url)
     }
```

Comment:
I think we can just do a `req.url.asString`

Reply:
If we use `req.url.asString` we will end up with an `url` instead of `uri`. I dont know if there would be any implications on using `url` instead

Reply:
@tusharmath I had a look at https://github.com/dream11/zio-http/issues/237 and it seems like using `req.url.asString` is not a viable solution, let me know what you think

Reply:
I think we can add a local method `makeUrl` or something better, inside of `EncodeRequest` to handle this specialized `string`.

Keeping `asUriString` inside of `URL` is a bit confusing considering we already have `asString` method on URL.

Reply:
I thought about doing this in the first place, but that would make us end up with this bit of logic copy and pasted in 2 different locations
```scala
    val encoder = new JQueryStringEncoder(url.path.asString)
    url.queryParams.foreach { case (key, values) =>
      values.foreach { value => encoder.addParam(key, value) }
    }
    encoder.toString
```
The best alternative to copy/paste would be to rename the `URL#path` to possibly `URL#uri`, make it public and return an `URI`.
This way we can call this method to from `EncodeRequest` to get the `Uri` and then call `.toString` and get the desired result.
I will push this change in a dedicated commit so you can have a look and let me know if its a good solution

Reply:
We could also add a method on URL called `queryString` which would contain the logic for building a string using the query params.

Reply:
Alternatively, we can add a method `relative` which always returns a relative url and then call `asString` on it?

Reply:
> We could also add a method on URL called `queryString` which would contain the logic for building a string using the query params.
> Then inside the encode use `path.asString + url.queryString` to produce the final url string.

I think the issue with this is that we are using the `JQueryStringEncoder` to deal with appending the query params, which requires a `uri` string to be created.
We could just manually write the conversion from the queryParams `Map[String, List[String]]` to a `String` but now we would be left with 2 ways of encoding the query_params

---
> Alternatively, we can add a method `relative` which always returns a relative url and then call `asString` on it?

I can convert the result of `JQueryStringEncoder` to a `zhttp.URL` which I think will be the cleanest solution

Reply:
@tusharmath Have gone with the `relative` method as it is the simplest and cleanest solution in my opinion